Redid Config to be the same as Event

DalwynWasTaken opened the pull request:
Made Scoreboard and Tablist update on call for data, might be changed to have it load constantly and get data, depends on query amount

Renamed FrameEvent to RenderEvent

Ascynx commented on src/main/java/mod/gate/core/config/ConfigManager.java:
```diff
@@ -0,0 +1,119 @@
+        CONFIG_FILE.mkdirs();
+
+        assert MinecraftClient.getInstance().player != null;
+        USER_FILE = new File(CONFIG_FILE, MinecraftClient.getInstance().player.getUuid() + FILE_SUFFIX);
```
I'm not sure when you want to run this method but if it's ran during the mod initialization it would crash since player isn't initialized yet.

DalwynWasTaken replied:
Right. But there is no other way to get UUID or how do you want to store player data

Ascynx replied:
I'm currently looking to see if there's a solution to be able to load it after the player is initialized.

Ascynx replied:
got it

Ascynx replied:
see https://github.com/Cobble8/Gate/pull/6#discussion_r917621004

DalwynWasTaken replied:
A solution or impossible. I have a different solution. Config or default

Ascynx commented on src/main/java/mod/gate/features/HudTesting.java:
```diff
@@ -1,22 +1,24 @@
     private final Identifier github = new Identifier(Reference.MODID, "textures/github.png");
 
+    @Config(description = "A random config value that will be written to config file")
+    public final boolean RandomConfigOption = true;
```
this isn't used by anything in the mod ?

DalwynWasTaken replied:
Correct. It’s an example of a config. I still need to make the gui which will build of the description part

Ascynx replied:
kk, should prolly just be labeled as an example

DalwynWasTaken replied:
Can you not tell from description? Alr I'll add that

Ascynx replied:
it's always better to be clear

Ascynx commented on src/main/resources/gate.mixins.json:
```diff
@@ -6,7 +6,7 @@
   "mixins": [
```
I think we should move mixins from ``"mixins": []`` to ``"client": []`` to avoid them injecting if the mod is loaded server side

Ascynx replied:
(this is only a suggestion)

DalwynWasTaken replied:
Sure. I don’t know the difference so I don’t mind as long as it works

Ascynx commented on src/main/java/mod/gate/core/config/ConfigManager.java:
```diff
@@ -0,0 +1,119 @@
+        CONFIG_FILE.mkdirs();
+
+        assert MinecraftClient.getInstance().player != null;
+        USER_FILE = new File(CONFIG_FILE, MinecraftClient.getInstance().player.getUuid() + FILE_SUFFIX);
```
```suggestion
        USER_FILE = new File(CONFIG_FILE, MinecraftClient.getInstance().getSession().getUuid() + FILE_SUFFIX);
```

Ascynx replied:
and remove the player assertion

DalwynWasTaken replied:
Oh this is the solution

DalwynWasTaken committed:
Redid Config to be the same as Event

Made Scoreboard and Tablist update on call for data, might be changed to have it load constantly and get data, depends on query amount

Renamed FrameEvent to RenderEvent

Ascynx reviewed (approved):
I don't see anything else that needs modification
